File: src/schedule_search.py
def print_info(schedule, members):
    date, stage, team = schedule['date'], schedule['stage'], schedule['team']
    members = ", ".join(members)
    info = 'Date   : %s\nStage  : %s\nTeam   : %s\nMembers: %s' % (date, stage, team, members)
    return info


def search_musician(name, schedule, members):
    out = [print_info(schedule.iloc[i], m) for i, m in enumerate(members) if name in m]
    out = "\n\n".join(out)
    return out

File: src/schedule_search.R
print_info <- function(performance) {
  members <- 
    performance$members %>% 
    unlist() %>% 
    unname() %>% 
    str_c(collapse = ", ")
      
  str_c("Date   : ", performance$date, "\n",
        "Stage  : ", performance$stage, "\n",
        "Team   : ", performance$team, "\n",
        "Members: ", members, "\n")
}

search_musician <- function(name, schedule) {
  idx <- sapply(schedule$members, function(x) {any(name %in% x)})
  if (!any(idx)) {  # All FALSE
    "없졍"
  } else {
    schedule <- schedule %>% filter(idx)
    
    info <- ""
    for (i in 1:nrow(schedule)) {
      newline <- print_info(schedule[i, ])
      info <- str_c(info, newline, sep = "\n")
    }
    info
  }
}
